sdk/storage/azure-storage-blob/azure/storage/blob/container_client.py

```diff
@@ -747,6 +759,7 @@ def walk_blobs(
             marker=marker,
             delimiter=delimiter)
 
+    @distributed_trace
     def upload_blob(
```

Comment:
This function calls into the BlobClient.upload function which you have decorated separately - does that matter?

Reply:
ok so I talked to Johan and it is his opinion that there will need to be some refactoring in the future: #6492